plugins/system_upgrade.py

```diff
@@ -626,14 +650,43 @@ def run_log(self):
         else:
             list_logs()
 
+    # == resolved_*: do staff after succesful resolvement =====================
+
+    def resolved_upgrade(self):
+        """Adjust transaction reasons according to stored values"""
+        if not self.cli.base.transaction:
+            return
+        install_packages = self.state.install_packages
+        remove_packages = self.state.remove_packages
+        for tsi in self.cli.base.transaction:
+            if tsi.action in dnf.transaction.FORWARD_ACTIONS:
+                try:
+                    pkg = tsi.pkg
+                    try:
+                        stored_reason = install_packages[pkg.repo.id][str(pkg)][str(tsi.action)]
+                        if stored_reason != tsi.reason:
+                            tsi.reason = stored_reason
+                    except KeyError:
+                        pass
+                except KeyError:
+                    raise RuntimeError("TransactionItem is has no RPM attached: %s" % tsi)
+            if tsi.action in dnf.transaction.BACKWARD_ACTIONS + \
+                    [libdnf.transaction.TransactionItemAction_REINSTALLED]:
+                try:
+                    pkg = tsi.pkg
+                    try:
+                        stored_reason = remove_packages[str(pkg)][str(tsi.action)]
+                        if stored_reason != tsi.reason:
+                            tsi.reason = stored_reason
+                    except KeyError:
+                        pass
+                except KeyError:
+                    raise RuntimeError("TransactionItem is has no RPM attached: %s" % tsi)
```

Comment:
This error is on more places than just the one and I suppose it's the same case.